.github/workflows/code-style-fix.yml

```diff
@@ -0,0 +1,41 @@
+name: Auto Code-Style Fix
+
+on: [pull_request, pull_request_target]
+
+jobs:
+  code_style_fix:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout 🛒️
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Black ◼️
+        uses: psf/black@stable
+        with:
+          options: >
+            --line-length=120
+            --skip-magic-trailing-comma
+            --target-version=py38
+            chat eval finetune generate lit_gpt notebooks pretrain quantize scripts tests xla
+          jupyter: true
+
+      - name: Ruff ®️
+        uses: chartboost/ruff-action@v1
+        with:
+          args: >
+            --select E,W,F,S
+            --extend-select C4,SIM,RET,PT,I001,ANN001,ANN201,ANN205,ANN206,ARG001
+            --ignore E501,E731,S108,S101,S113,S603,PT007,S310,E402,PT004,C408
+            --per-file-ignores "*/**/__init__.py":I001,"tests/*":ANN
+            --line-length 120
+            --target-version py38
+            --fix-only
+
+      - name: Fixing Pull Request 🛠️
+        uses: actions-js/push@v1.4
+        with:
+          branch: ${{ github.head_ref || github.ref_name }}
+          github_token: ${{ secrets.PAT_GHOST }}
```

Comment:
I don't think this works if the PR comes from a fork

Reply:
Only use `pull_request_target` but that won't likely push to forked PRs, so the best way would be to install precommit bot if you really want also fixing... Otherwise our utilities repo has precommit workflow for reuse

Reply:
> When I did it in my fork of the repo, I did these steps (maybe you've missed one)

That is correct, GH doesn't not share secrets fork PRs by design

Reply:
> Plus I don't see how pre-commit would work differently here, as the issue is in the way we push changes back.

Bot is third party actor, so it behave identical for main repo development as well as forked contributing

Reply:
I believe Jirka is right here and any action-based solution is doomed since PATs cannot be shared with forks with `pull_request` events.

Reply:
I never expected that a PAT (or more specifically a secret tied to the PAT) is shared with forks. I guess that's why they are called a **secret** and a **personal** token.
That's why I provide in the workflow two tokens to choose from: the secret and the default github token.
```yaml
github_token: ${{ secrets.action_secret || github.token }}
```
so if there is no secret with name `action_secret` then the `github.token` is used.

The only difference in the workflow will be that with the secret all the running workflows will be restarted if there are any changes were made after linting. With the default token they will be running in parallel (on outdated commit).

Reply:
@Borda You are talking about this pre-commit bot: https://pre-commit.ci/ ?
> configuration: zero configuration setup -- nothing is needed beyond the .pre-commit-config.yaml file you already have!

And for this repo this config file is *no bueno*.